js/src/ui/Form/AddressSelect/addressSelectStore.js

```diff
@@ -194,6 +239,8 @@ export default class AddressSelectStore {
           .filter((result) => result && !ZERO.test(result.address));
```

Comment:
This is dangerous, especially when we have (or add in the future) long-running queries, as it will show incorrect values to the users for a while. I'd rather introduce a loading flag.